### ISSUE TITLE
`azurerm_pim_active_role_assignment` failing to create beyond version 1.104.0

### ISSUE THREAD

**Issue by alexwilcox9** [hashicorp/terraform-provider-azurerm#26023]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.8.3

### AzureRM Provider Version

3.104.0

### Affected Resource(s)/Data Source(s)

azurerm_pim_active_role_assignment, azurerm_pim_eligible_role_assignment

### Terraform Configuration Files

```hcl
data "azurerm_client_config" "example" {}

resource "azurerm_resource_group" "example" {
  name     = "example-rg"
  location = "UK South"
}

data "azurerm_role_definition" "example" {
  scope = azurerm_resource_group.example.id
  name  = "Owner"
}

resource "azurerm_pim_active_role_assignment" "example" {
  scope              = azurerm_resource_group.example.id
  role_definition_id = data.azurerm_role_definition.example.id
  principal_id       = data.azurerm_client_config.example.object_id
}


provider "azurerm" {
  features {}
}

terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "= 3.104"
    }
  }
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: waiting for Scoped Role Assignment Schedule Request (Scope: "/subscriptions/<SUB-ID>/resourceGroups/example-rg"
│ Role Assignment Schedule Request Name: "128c3ef0-9f5d-1d3c-66e6-faa8baebe826") to be created: creating Scoped Role Assignment Schedule Request (Scope: "/subscriptions/<SUB-ID>/resourceGroups/example-rg"
│ Role Assignment Schedule Request Name: "128c3ef0-9f5d-1d3c-66e6-faa8baebe826"): unexpected status 400 (400 Bad Request) with error: InvalidRoleAssignmentRequestSchedule: The role assignment request schedule is invalid.
│ 
│   with azurerm_pim_active_role_assignment.example,
│   on main.tf line 13, in resource "azurerm_pim_active_role_assignment" "example":
│   13: resource "azurerm_pim_active_role_assignment" "example" {
│ 
│ waiting for Scoped Role Assignment Schedule Request (Scope: "/subscriptions/<SUB-ID>/resourceGroups/example-rg"
│ Role Assignment Schedule Request Name: "128c3ef0-9f5d-1d3c-66e6-faa8baebe826") to be created: creating Scoped Role Assignment Schedule Request (Scope:
│ "/subscriptions/<SUB-ID>/resourceGroups/example-rg"
│ Role Assignment Schedule Request Name: "128c3ef0-9f5d-1d3c-66e6-faa8baebe826"): unexpected status 400 (400 Bad Request) with error: InvalidRoleAssignmentRequestSchedule: The role assignment request schedule is
│ invalid.
```

### Expected Behaviour

The role assignment should create a permanent active assignment of the role
When updating from previous provider versions nothing should happen

### Actual Behaviour

The assignment fails to create with the above error
Also if you have created the role assignments previously in an older provider version they will be flagged for replacement after updating to 3.104.0

```
  # azurerm_pim_active_role_assignment.example must be replaced
-/+ resource "azurerm_pim_active_role_assignment" "example" {
      ~ id                 = "/subscriptions/<SUB-ID>/resourceGroups/example-rg|/subscriptions/<SUB-ID>/providers/Microsoft.Authorization/roleDefinitions/8e3af657-a8ff-443c-a75c-2fe8c4bcb635|4b1d0582-af47-411b-bc98-c6c7d0bde959" -> (known after apply)
      + justification      = (known after apply)
      ~ principal_type     = "User" -> (known after apply)
        # (3 unchanged attributes hidden)

      - schedule { # forces replacement
          - start_date_time = "2024-05-19T18:19:35.8962731+00:00" -> null

          - expiration {
              - duration_days  = 0 -> null
              - duration_hours = 0 -> null
                # (1 unchanged attribute hidden)
            }
        }

      - ticket { # forces replacement
            # (2 unchanged attributes hidden)
        }
    }

```

### Steps to Reproduce

1. `terraform apply`
2.  At this stage on either version it should fail due to the default settings on the owner role
   Enable `Allow permanent active assignment`
   Disable `Require justification on active assignment`
   Enable `Allow permanent eligible assignment` (when testing `azurerm_pim_eligible_role_assignment`)
3. Re run `terraform apply`

On version 3.103.0 this will succeed, on 3.104.0 it will fail (or if updating show the need to replace the resource)

### Important Factoids

If you destroy and re-apply the role config has to set again.

### References

https://github.com/hashicorp/terraform-provider-azurerm/pull/25956

**Comment by kkadiyam** [comment]
We are seeing same issue too, All existing PIM eligible assignments are getting replaced.

**Comment by kkadiyam** [comment]
New release 3.104.1 or 3.104.2 didn't fix our issue, We are still seeing existing eligible assignments getting recreated even-though the state has the assignments and the portal is in sync with state.

**Comment by manicminer** [comment]
Thanks for reporting @kkadiyam, we'll work on getting this resolved asap, please see #26050 for updates.
